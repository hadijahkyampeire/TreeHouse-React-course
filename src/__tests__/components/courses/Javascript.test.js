import React from 'react';
import enzyme from 'enzyme';
import Javascript from '../../../components/courses/JavaScript';

describe('About component', () => {
    const node = enzyme.shallow(<Javascript/>);

    it('should render without crashing', () => {
        expect(node).toMatchSnapshot();
    });
});