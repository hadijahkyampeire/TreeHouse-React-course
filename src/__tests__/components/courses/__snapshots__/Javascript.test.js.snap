// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JavaScript />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <Course
                desc="JavaScript is a programming language that drives the web: from front-end user interface design, to backend server-side programming, you'll find JavaScript at every stage of a web site and web application. In this course, you'll learn the fundamental programming concepts and syntax of the JavaScript programming language."
                img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage1.png"
                title="JavaScript Basics"
        />
        <Course
                desc="Storing, tracking and handling data is a large part of computer programming. Arrays provide a method for storing multiple values into a single variable. That makes an array a convenient way to pass around a list of items."
                img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage5.png"
                title="JavaScript Loops, Arrays and Objects"
        />
        <Course
                desc="jQuery Basics covers why you'd want to use jQuery, what it is and how to include it in your projects. You'll build several projects over the course to give you the confidence to integrate jQuery in your own projects and add that level of flair and interactivity to any site you work on."
                img="https://achievement-images.teamtreehouse.com/badges_JavaScript_jQueryBasics_Stage1.png"
                title="jQuery Basics"
        />
        <Course
                desc="AJAX is an important front-end web technology that lets JavaScript communicate with a web server. It lets you load new content without leaving the current page, creating a better, faster experience for your web site's visitors. In this course, you'll learn how AJAX works and how you can use JavaScript to communicate with a web server."
                img="https://achievement-images.teamtreehouse.com/badges_JavaScript_AJAXBasics_Stage3.png"
                title="AJAX Basics"
        />
        <Course
                desc="In this course we’ll create a to-do list application using JavaScript alone, without using any third party libraries. You’ll get to grips with manipulating and traversing the DOM and adding event handlers to web page elements. The things that you can do with JavaScript are always increasing and this course will give you a solid footing going forward."
                img="https://achievement-images.teamtreehouse.com/bagdes-javascript-interactiveweb-stage-12.png"
                title="Interactive Web Pages with JavaScript"
        />
        <Course
                desc="In this course we will create a command line application to retrieve user's profile information from the Treehouse website. We'll be writing our application in JavaScript to run on the Node.js platform."
                img="https://achievement-images.teamtreehouse.com/badges_JavaScript_nodeBasics_Stage1.png"
                title="Node.js Basics"
        />
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Course
            desc="JavaScript is a programming language that drives the web: from front-end user interface design, to backend server-side programming, you'll find JavaScript at every stage of a web site and web application. In this course, you'll learn the fundamental programming concepts and syntax of the JavaScript programming language."
            img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage1.png"
            title="JavaScript Basics"
/>,
          <Course
            desc="Storing, tracking and handling data is a large part of computer programming. Arrays provide a method for storing multiple values into a single variable. That makes an array a convenient way to pass around a list of items."
            img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage5.png"
            title="JavaScript Loops, Arrays and Objects"
/>,
          <Course
            desc="jQuery Basics covers why you'd want to use jQuery, what it is and how to include it in your projects. You'll build several projects over the course to give you the confidence to integrate jQuery in your own projects and add that level of flair and interactivity to any site you work on."
            img="https://achievement-images.teamtreehouse.com/badges_JavaScript_jQueryBasics_Stage1.png"
            title="jQuery Basics"
/>,
          <Course
            desc="AJAX is an important front-end web technology that lets JavaScript communicate with a web server. It lets you load new content without leaving the current page, creating a better, faster experience for your web site's visitors. In this course, you'll learn how AJAX works and how you can use JavaScript to communicate with a web server."
            img="https://achievement-images.teamtreehouse.com/badges_JavaScript_AJAXBasics_Stage3.png"
            title="AJAX Basics"
/>,
          <Course
            desc="In this course we’ll create a to-do list application using JavaScript alone, without using any third party libraries. You’ll get to grips with manipulating and traversing the DOM and adding event handlers to web page elements. The things that you can do with JavaScript are always increasing and this course will give you a solid footing going forward."
            img="https://achievement-images.teamtreehouse.com/bagdes-javascript-interactiveweb-stage-12.png"
            title="Interactive Web Pages with JavaScript"
/>,
          <Course
            desc="In this course we will create a command line application to retrieve user's profile information from the Treehouse website. We'll be writing our application in JavaScript to run on the Node.js platform."
            img="https://achievement-images.teamtreehouse.com/badges_JavaScript_nodeBasics_Stage1.png"
            title="Node.js Basics"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "js-1",
          "nodeType": "function",
          "props": Object {
            "desc": "JavaScript is a programming language that drives the web: from front-end user interface design, to backend server-side programming, you'll find JavaScript at every stage of a web site and web application. In this course, you'll learn the fundamental programming concepts and syntax of the JavaScript programming language.",
            "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage1.png",
            "title": "JavaScript Basics",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "js-2",
          "nodeType": "function",
          "props": Object {
            "desc": "Storing, tracking and handling data is a large part of computer programming. Arrays provide a method for storing multiple values into a single variable. That makes an array a convenient way to pass around a list of items.",
            "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage5.png",
            "title": "JavaScript Loops, Arrays and Objects",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "js-3",
          "nodeType": "function",
          "props": Object {
            "desc": "jQuery Basics covers why you'd want to use jQuery, what it is and how to include it in your projects. You'll build several projects over the course to give you the confidence to integrate jQuery in your own projects and add that level of flair and interactivity to any site you work on.",
            "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_jQueryBasics_Stage1.png",
            "title": "jQuery Basics",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "js-4",
          "nodeType": "function",
          "props": Object {
            "desc": "AJAX is an important front-end web technology that lets JavaScript communicate with a web server. It lets you load new content without leaving the current page, creating a better, faster experience for your web site's visitors. In this course, you'll learn how AJAX works and how you can use JavaScript to communicate with a web server.",
            "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_AJAXBasics_Stage3.png",
            "title": "AJAX Basics",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "js-5",
          "nodeType": "function",
          "props": Object {
            "desc": "In this course we’ll create a to-do list application using JavaScript alone, without using any third party libraries. You’ll get to grips with manipulating and traversing the DOM and adding event handlers to web page elements. The things that you can do with JavaScript are always increasing and this course will give you a solid footing going forward.",
            "img": "https://achievement-images.teamtreehouse.com/bagdes-javascript-interactiveweb-stage-12.png",
            "title": "Interactive Web Pages with JavaScript",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "js-6",
          "nodeType": "function",
          "props": Object {
            "desc": "In this course we will create a command line application to retrieve user's profile information from the Treehouse website. We'll be writing our application in JavaScript to run on the Node.js platform.",
            "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_nodeBasics_Stage1.png",
            "title": "Node.js Basics",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <Course
                    desc="JavaScript is a programming language that drives the web: from front-end user interface design, to backend server-side programming, you'll find JavaScript at every stage of a web site and web application. In this course, you'll learn the fundamental programming concepts and syntax of the JavaScript programming language."
                    img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage1.png"
                    title="JavaScript Basics"
          />
          <Course
                    desc="Storing, tracking and handling data is a large part of computer programming. Arrays provide a method for storing multiple values into a single variable. That makes an array a convenient way to pass around a list of items."
                    img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage5.png"
                    title="JavaScript Loops, Arrays and Objects"
          />
          <Course
                    desc="jQuery Basics covers why you'd want to use jQuery, what it is and how to include it in your projects. You'll build several projects over the course to give you the confidence to integrate jQuery in your own projects and add that level of flair and interactivity to any site you work on."
                    img="https://achievement-images.teamtreehouse.com/badges_JavaScript_jQueryBasics_Stage1.png"
                    title="jQuery Basics"
          />
          <Course
                    desc="AJAX is an important front-end web technology that lets JavaScript communicate with a web server. It lets you load new content without leaving the current page, creating a better, faster experience for your web site's visitors. In this course, you'll learn how AJAX works and how you can use JavaScript to communicate with a web server."
                    img="https://achievement-images.teamtreehouse.com/badges_JavaScript_AJAXBasics_Stage3.png"
                    title="AJAX Basics"
          />
          <Course
                    desc="In this course we’ll create a to-do list application using JavaScript alone, without using any third party libraries. You’ll get to grips with manipulating and traversing the DOM and adding event handlers to web page elements. The things that you can do with JavaScript are always increasing and this course will give you a solid footing going forward."
                    img="https://achievement-images.teamtreehouse.com/bagdes-javascript-interactiveweb-stage-12.png"
                    title="Interactive Web Pages with JavaScript"
          />
          <Course
                    desc="In this course we will create a command line application to retrieve user's profile information from the Treehouse website. We'll be writing our application in JavaScript to run on the Node.js platform."
                    img="https://achievement-images.teamtreehouse.com/badges_JavaScript_nodeBasics_Stage1.png"
                    title="Node.js Basics"
          />
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Course
              desc="JavaScript is a programming language that drives the web: from front-end user interface design, to backend server-side programming, you'll find JavaScript at every stage of a web site and web application. In this course, you'll learn the fundamental programming concepts and syntax of the JavaScript programming language."
              img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage1.png"
              title="JavaScript Basics"
/>,
            <Course
              desc="Storing, tracking and handling data is a large part of computer programming. Arrays provide a method for storing multiple values into a single variable. That makes an array a convenient way to pass around a list of items."
              img="https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage5.png"
              title="JavaScript Loops, Arrays and Objects"
/>,
            <Course
              desc="jQuery Basics covers why you'd want to use jQuery, what it is and how to include it in your projects. You'll build several projects over the course to give you the confidence to integrate jQuery in your own projects and add that level of flair and interactivity to any site you work on."
              img="https://achievement-images.teamtreehouse.com/badges_JavaScript_jQueryBasics_Stage1.png"
              title="jQuery Basics"
/>,
            <Course
              desc="AJAX is an important front-end web technology that lets JavaScript communicate with a web server. It lets you load new content without leaving the current page, creating a better, faster experience for your web site's visitors. In this course, you'll learn how AJAX works and how you can use JavaScript to communicate with a web server."
              img="https://achievement-images.teamtreehouse.com/badges_JavaScript_AJAXBasics_Stage3.png"
              title="AJAX Basics"
/>,
            <Course
              desc="In this course we’ll create a to-do list application using JavaScript alone, without using any third party libraries. You’ll get to grips with manipulating and traversing the DOM and adding event handlers to web page elements. The things that you can do with JavaScript are always increasing and this course will give you a solid footing going forward."
              img="https://achievement-images.teamtreehouse.com/bagdes-javascript-interactiveweb-stage-12.png"
              title="Interactive Web Pages with JavaScript"
/>,
            <Course
              desc="In this course we will create a command line application to retrieve user's profile information from the Treehouse website. We'll be writing our application in JavaScript to run on the Node.js platform."
              img="https://achievement-images.teamtreehouse.com/badges_JavaScript_nodeBasics_Stage1.png"
              title="Node.js Basics"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "js-1",
            "nodeType": "function",
            "props": Object {
              "desc": "JavaScript is a programming language that drives the web: from front-end user interface design, to backend server-side programming, you'll find JavaScript at every stage of a web site and web application. In this course, you'll learn the fundamental programming concepts and syntax of the JavaScript programming language.",
              "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage1.png",
              "title": "JavaScript Basics",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "js-2",
            "nodeType": "function",
            "props": Object {
              "desc": "Storing, tracking and handling data is a large part of computer programming. Arrays provide a method for storing multiple values into a single variable. That makes an array a convenient way to pass around a list of items.",
              "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_Basics_Stage5.png",
              "title": "JavaScript Loops, Arrays and Objects",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "js-3",
            "nodeType": "function",
            "props": Object {
              "desc": "jQuery Basics covers why you'd want to use jQuery, what it is and how to include it in your projects. You'll build several projects over the course to give you the confidence to integrate jQuery in your own projects and add that level of flair and interactivity to any site you work on.",
              "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_jQueryBasics_Stage1.png",
              "title": "jQuery Basics",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "js-4",
            "nodeType": "function",
            "props": Object {
              "desc": "AJAX is an important front-end web technology that lets JavaScript communicate with a web server. It lets you load new content without leaving the current page, creating a better, faster experience for your web site's visitors. In this course, you'll learn how AJAX works and how you can use JavaScript to communicate with a web server.",
              "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_AJAXBasics_Stage3.png",
              "title": "AJAX Basics",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "js-5",
            "nodeType": "function",
            "props": Object {
              "desc": "In this course we’ll create a to-do list application using JavaScript alone, without using any third party libraries. You’ll get to grips with manipulating and traversing the DOM and adding event handlers to web page elements. The things that you can do with JavaScript are always increasing and this course will give you a solid footing going forward.",
              "img": "https://achievement-images.teamtreehouse.com/bagdes-javascript-interactiveweb-stage-12.png",
              "title": "Interactive Web Pages with JavaScript",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "js-6",
            "nodeType": "function",
            "props": Object {
              "desc": "In this course we will create a command line application to retrieve user's profile information from the Treehouse website. We'll be writing our application in JavaScript to run on the Node.js platform.",
              "img": "https://achievement-images.teamtreehouse.com/badges_JavaScript_nodeBasics_Stage1.png",
              "title": "Node.js Basics",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
